### PR TITLE
Write hyperspectral data

### PR DESCRIPTION
**Pull request by JorgeGtz**
**Describe your changes**
Add the functionality to write ENVI-format hyperspectral images.

**Type of update**
Is this a:
* New feature or feature enhancement

**Associated issues**

**Additional context**

### CODE REVIEW

File: plantcv/plantcv/hyperspectral/write_data.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import os
+import numpy as np
+
+from plantcv.plantcv.__init__ import __version__
```
Comment (nfahlgren) [inline plantcv/plantcv/hyperspectral/write_data.py:4]:
```suggestion
from plantcv.plantcv import __version__
```

File: tests/plantcv/hyperspectral/test_write_data.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+                                filename='random_hyperspectral_test',
+                                default_bands=None)
+
+
+    filename = os.path.join(cache_dir, 'plantcv_hyperspectral_write_data.raw')
+    out = write_data(filename=filename, spectral_data=rand_spectral_array)
+
+    # Read written hyperspectral image
+    array_data = read_data(filename=filename)
+    assert np.shape(array_data.array_data) == (lines, samples, bands)
```
Comment (jdavidpeery) [inline tests/plantcv/hyperspectral/test_write_data.py:52]: Would going beyond checking if the dimensions are the same to see if some of the values are the same be helpful? Or is just checking that they're the same shape usually sufficient

Comment (jdavidpeery) [inline tests/plantcv/hyperspectral/test_write_data.py:52]: Would going beyond checking if the dimensions are the same to see if some of the values are the same be helpful? Or is just checking that they're the same shape usually sufficient

File: docs/write_data.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
+## Write data
+
+Write a hyperspectral image in ENVI format to the specified file.
+It creates a text header file with extension .hdr and a binary file with
+extension .raw.  This function only supports Band-interleaved-by-line (BIL)
+interleave.
+
+**plantcv.hyperspectral.write_data**(*filename, spectral_data*):
+
+**returns** Bool
```
[the remark this answers is not in the document]
Comment (JorgeGtz) [reply docs/write_data.md:10]: I think we can just eliminate the output to be coherent with the other writing functions

File: docs/write_data.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
+## Write data
+
+Write a hyperspectral image in ENVI format to the specified file.
+It creates a text header file with extension .hdr and a binary file with
+extension .raw.  This function only supports Band-interleaved-by-line (BIL)
```
[the remark this answers is not in the document]
Comment (jdavidpeery) [reply docs/write_data.md:5]:
```suggestion
extension .raw.  This function only supports Band-Interleaved-by-Line (BIL)
```

File: docs/write_data.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (jdavidpeery) [reply docs/write_data.md:10]:
```suggestion

```

File: plantcv/plantcv/hyperspectral/write_data.py [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+        for wl in wavelenghths[:-1]:
+            f.write(f'{wl},\n')
+        f.write(f'{wavelenghths[-1]}\n')
+        f.write('}')
+
+    # create raw binary file containing the hyperspectral array values
+    with open(filename+'.raw', mode='w+b') as f:
+        f.write(spectral_data.array_data.transpose(0,2,1).tobytes(order='C'))
+
+    return True
```
Comment (jdavidpeery) [inline plantcv/plantcv/hyperspectral/write_data.py:48]:
```suggestion
```

File: tests/plantcv/hyperspectral/test_write_data.py [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+                                interleave='bil',
+                                wavelength_units='nm',
+                                array_type="datacube",
+                                pseudo_rgb=None,
+                                filename='random_hyperspectral_test',
+                                default_bands=None)
+
+
+    filename = os.path.join(cache_dir, 'plantcv_hyperspectral_write_data.raw')
+    out = write_data(filename=filename, spectral_data=rand_spectral_array)
```
Comment (jdavidpeery) [inline tests/plantcv/hyperspectral/test_write_data.py:48]:
```suggestion
    write_data(filename=filename, spectral_data=rand_spectral_array)
```

File: tests/plantcv/hyperspectral/test_write_data.py (hunk 2 above)
Comment (jdavidpeery) [reply tests/plantcv/hyperspectral/test_write_data.py:52]: Ok cool, would just a simple `assert np.array_equal(array_data.array_data, rand_spectral_array)` be good or do you think that would take up more time than it's worth?

Comment (JorgeGtz) [reply tests/plantcv/hyperspectral/test_write_data.py:52]: I think both ways are good options

**Review by jdavidpeery — APPROVED** [review]
Looks good then

**Pull request merged**